VogtPayroll2/Employee1.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtPayroll2
+{
+    abstract class Employee
+    {
+        //For simplicity, use a constant tax rate of 30% to compute the tax amount
+        protected string name;
+        protected int hoursWorked;
+        protected decimal payRate;
+        protected const decimal taxAmount = .30m;
+
+        public Employee(string aName, int aHoursWorked, decimal aPayRate)
+        {
+            name = aName;
+            hoursWorked = aHoursWorked;
+            payRate = aPayRate;
+        }
+        public void DisplayEmployeeInfo(Employee employee)
+        {
+            // The output should display the aName of each Employee, hours worked, hourly rate, overtime pay, regular (gross) pay, tax amount, and net pay.
+            Console.WriteLine($"Employee Name: {employee.name}");
+            Console.WriteLine($"Hours worked: {employee.hoursWorked}");
+            Console.WriteLine($"Hourly rate: {GetPayRate()}");
+            Console.WriteLine($"Overtime pay: {OverTime()}");
+            Console.WriteLine($"Regular (gross) pay: {GetGrossPay()}");
+            Console.WriteLine($"Tax amount: {GetTaxAmount()}");
+            Console.WriteLine($"Net Pay: {GetNetPay()}");
+        }
+
+        public void DisplayNetPayOfAllEmployees(Employee[] employee)
+        {
+            // The average net pay of all employees should also be displayed
+        }
+
+        public decimal FindMinimumAndMaximumNetPay()
+        {
+            // In addition, the program should find the minimum and maximum net pay of all employees as well as sort the employees based on their net pay (ascending order)
+            // The program must combine two sorting techniques; the Selection and Exchange sorts. It will efficiently sort the employees' net pay.
+            // Preferrably without using a data file.
+            decimal netPay = default;
+
+            return netPay;
+        }
+
+        public decimal OverTime()
+        {
+            //Employees that work over 40 hours will receive overtime pay of one and a half of their hourly rate for overtime hours worked.
+            decimal overTimePay = default;
+
+            overTimePay = Pay(payRate);
+
+            return overTimePay;
+        }
+
+        public decimal Pay(decimal payRate)
```

Comment:
Hard to tell what this function is doing by the name. Is this really what is calculating the overtime pay? If so theres no reason to have another function which simply returns this function. 

The payrate is already assigned to the class during initialization (via the constructor), so you don't need to pass it in.
```
 To compute the overtime pay for a salary based employee, first find the hourly rate by dividing the gross pay by 40, and then compute overtime pay.
```

With that in mind. I think you need to add 1 more abstract faction named GetOvertimePayRate. Then have it implemented different for salary vs hourly.

Then, this `GetOvertimePay()` should look more like:

```c#
if (hoursWorked < 40)
  return 0;

return GetOvertimePayRate() * (hoursWorked - 40);
```